modern/src/LanguageProvider.js

```diff
@@ -0,0 +1,18 @@
+import {LanguageContext} from "./LanguageContext";
+import React, {useState} from "react";
+import {defaultLanguage, setSelectedLanguage} from "./common/localization";
+
+const LanguageProvider = (props) => {
+    const [language, setLanguage] = useState(defaultLanguage);
+
+    const handleLanguageChange = (nextLanguage) => {
+        setSelectedLanguage(nextLanguage);
+        setLanguage(nextLanguage);
+    };
+
+    return <LanguageContext.Provider value={{language, setLanguage: handleLanguageChange}}>
+        {props.children}
+    </LanguageContext.Provider>
```

Comment:
```suggestion
    return (
      <LanguageContext.Provider value={{language, setLanguage: handleLanguageChange}}>
        {props.children}
      </LanguageContext.Provider>
    )
```

Reply:
btw please use 2-spaces indentation everywhere. Make sure the style is consistent with the rest of the code.